flow/src/main/java/com/sequenceiq/flow/reactor/api/event/BaseFlowEvent.java

```diff
@@ -35,7 +35,7 @@ public BaseFlowEvent(
         this.selector = selector;
         this.resourceId = resourceId;
         this.resourceCrn = resourceCrn;
-        this.accepted = accepted;
+        this.accepted = Objects.requireNonNullElse(accepted, new Promise<>());
```

Comment:
I don't see clearly why is it needed. If you want to explicitly pass null, use the constructor above.

Reply:
as discussed, it was needed so that event builders do not have to handle the default value